src/net.cpp

```diff
@@ -1010,17 +1010,25 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     // on all platforms.  Set it again here just to be sure.
     SetSocketNoDelay(hSocket);
 
-    int bannedlevel = m_banman ? m_banman->IsBannedLevel(addr) : 0;
+    bool banned = m_banman->IsBanned(addr);
+    bool discouraged = banned || m_banman->IsDiscouraged(addr);
```

Comment:
or'ing with `banned` is confusing here. After this PR, banning and discouraging are totally separated concepts, but this code signals that discouraging is a superset of banning.

Reply:
I was trying to avoid a call to IsDiscouraged if the peer is already banned. I moved it to accomplish something similar. I think it's clearer now.